product/src/main/java/com/yas/product/repository/BrandRepository.java

```diff
@@ -2,11 +2,15 @@
 
 import com.yas.product.model.Brand;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
 import org.springframework.stereotype.Repository;
 
 import java.util.Optional;
 
 @Repository
 public interface BrandRepository extends JpaRepository<Brand,Long> {
     Optional<Brand> findBySlug(String slug);
+
+    @Query("select e from Brand e where e.name = ?1 and (?2 is null or e.id != ?2)")
```

Comment:
find by Id but i see e.id != ?2
please check

Reply:
Thanks Toan. I will rename method to clearly logic